### PR TITLE
Remove setting package file_type as a boolean in ROS discovery plugin.

### PR DESCRIPTION
**Pull request by tdenewiler**

### CODE REVIEW

File: tests/plugins/discovery/ros_discovery_plugin/test_ros_discovery_plugin.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def test_ros_discovery_plugin_scan_valid():
     )
     os.environ["ROS_VERSION"] = "1"
     rdp.scan(package, "level")
-    assert package["ros"]
+    assert "is_ros1" in package
```
Comment (xydesa) [inline tests/plugins/discovery/ros_discovery_plugin/test_ros_discovery_plugin.py:77]:
```suggestion
    assert "is_ros1" in package
    assert "is_ros2" not in package
```

File: tests/plugins/discovery/ros_discovery_plugin/test_ros_discovery_plugin.py [hunk 2]
```diff
@@ -152,7 +155,7 @@ def test_ros_discovery_plugin_ros2_scan_valid():
         os.environ["ROS_VERSION"] = "2"
         os.environ["ROS_DISTRO"] = "foxy"
         rdp.scan(package, "level")
-        assert package["ros"]
+        assert "is_ros2" in package
```
Comment (xydesa) [inline tests/plugins/discovery/ros_discovery_plugin/test_ros_discovery_plugin.py:158]:
```suggestion
        assert "is_ros2" in package
        assert "is_ros1" not in package
```

File: tests/plugins/discovery/ros_discovery_plugin/test_ros_discovery_plugin.py [hunk 3]
```diff
@@ -164,5 +167,4 @@ def test_ros_discovery_plugin_ros1_scan_valid():
     )
     os.environ["ROS_VERSION"] = "1"
     rdp.scan(package, "level")
-    assert package["ros"]
-    assert package["catkin"]
+    assert "is_ros1" in package
```
Comment (xydesa) [inline tests/plugins/discovery/ros_discovery_plugin/test_ros_discovery_plugin.py:170]:
```suggestion
    assert "is_ros1" in package
    assert "is_ros2" not in package
```

File: requirements.txt [hunk 4]
```diff
@@ -1,6 +1,7 @@
 PyYAML
 bandit
 black
+catkin-pkg
```
[the remark this answers is not in the document]
Comment (tdenewiler) [reply requirements.txt]: [c330a82](https://github.com/sscpac/statick/pull/439/commits/c330a8234cbbd2865debbc1a4bdb6fa73f20580e). I agree with you.

File: statick_tool/plugins/discovery/ros_discovery_plugin.py [hunk 5]
```diff
@@ -50,9 +50,8 @@ def scan(
             and ros_version is not None
         ):
             logging.info("  Package is ROS%s.", ros_version)
-            package["ros"] = True
             if ros_version == "1":
-                package["catkin"] = True
+                package["is_ros1"] = True
```
[the remark this answers is not in the document]
Comment (tdenewiler) [reply statick_tool/plugins/discovery/ros_discovery_plugin.py:54]: I added a comment in the linked issue that hopefully clears up the root cause that led to Statick crashes and adds motivation for these changes.

[the remark this answers is not in the document]
Comment (xydesa) [reply statick_tool/plugins/discovery/ros_discovery_plugin.py:54]: thanks, that helps a lot! i'm surprised my grep didn't help me find the catkin_lint_tool.

**Review by xydesa — APPROVED** [review]

**Pull request merged**